### PR TITLE
Lifecycle Graph

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- adds lifecycle statistic to trends
- majority of the lines are tests and raw SQL
*If this affects the front-end, include screenshots.*  
Screen Shot 2020-12-01 at 9 24 04 PM

## Checklist

- [ ] All querysets/queries filter by Organization, Team, and User (if this PR affects ANY querysets/queries).
- [ ] Django backend tests (if this PR affects the backend).
- [ ] Cypress end-to-end tests (if this PR affects the frontend).
- [ ] remove duplicate logic

### CODE REVIEW

File: ee/clickhouse/sql/trends/lifecycle.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+LIFECYCLE_SQL = """
+SELECT groupArray(day_start), groupArray(counts), status FROM (
```
Comment (EDsCODE) [inline ee/clickhouse/sql/trends/lifecycle.py:2]: Benchmark note: 2.5s across all time monthly interval on $pageviews

**Comment by EDsCODE** [comment]
Update: Lifecycle works! This PR still needs the logic for returning people when a bar is clicked on so that will be completed when I'm back. #2405 should be reviewed and merged first so I can update some of the param handling here to use the clearer format.

File: frontend/src/scenes/insights/BreakdownFilter.js [hunk 2]
```diff
@@ -147,7 +147,7 @@ export function BreakdownFilter({ filters, onChange }) {
                 <Button
                     shape="round"
                     type={breakdown ? 'primary' : 'default'}
-                    disabled={shown_as == 'Stickiness'}
+                    disabled={shown_as == 'Stickiness' || shown_as == 'Lifecycle'}
```
Comment (timgl) [inline frontend/src/scenes/insights/BreakdownFilter.js:150]: I can still click it despite being disabled

**Review by timgl — COMMENTED** [review]
Looks like on my local instance here I'm in both `resurrecting` and `new` (expect to only be in resurrecting) in the second to last week, and `returning` and `dormant` (expect to be retruning) in the last week.
image

Total volume/Active users dropdown etc still work but don't change the graph. Not sure it's worth disabling.

`Compare to previous` does kind of work but is a bit chaotic. Not sure what that should look like so might be worth disabling as well?

**Comment by EDsCODE** [comment]
@timgl when you click on the ressurecting/new bars does it show you there twice

**Comment by timgl** [comment]
Just once everywhere

**Comment by EDsCODE** [comment]
@timgl that problem above is good for another look! 

one more null state configuration that we should decide on is the number of entities that someone can add. I dont think the stacked charts make much sense if you have more than one entity but what do you think about the UX where we only keep the first entity when someone switches to lifecycle

File: frontend/src/scenes/insights/BreakdownFilter.js (hunk 2 above)
Comment (EDsCODE) [reply frontend/src/scenes/insights/BreakdownFilter.js:150]: fixed!

**Review by timgl — COMMENTED** [review]
- The lifecycle data looks correct!
- I think getting rid of the other entities does make sense to me. It does make me wonder whether lifecycle just needs to be another tab instead but not sure.
- Found a very weird bug just opening insights with nothing set. Master:
image

This PR:
image

**Comment by EDsCODE** [comment]
addressed the points above!

**Comment by timgl** [comment]
SOLID. let's quickly check it on app.posthog.net before deploying to heroku

**Pull request merged**